auth/authenticate.go

```diff
@@ -0,0 +1,141 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	"github.com/dgrijalva/jwt-go"
+)
+
+const auth0ApiIdentifier = "matterhorn-api"
+const auth0Domain = "https://matterhorn-prototype.auth0.com/"
+
+// getPemCerts gets the remote JWKS for the Auth0 account and returns the certificate with the public key in
+// PEM format
+func getPemCert(token *jwt.Token) (string, error) {
```

Comment:
what is Pem?

Reply:
Something Encryption Something. This is copy-pasted from Auth0, will find out and add more useful comments...